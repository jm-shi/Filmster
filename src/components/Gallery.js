import React from 'react';

import { connect } from 'react-redux';
import { fetchMovieDetails } from '../actions/movieActions';
import { history } from '../AppRouter';

class Gallery extends React.Component {
  goToDetails = id => {
    if (id) {
      history.push({
        pathname: '/details',
        state: id
      });
    }
  };
  render() {
    const poster_path = 'https://image.tmdb.org/t/p/w185/';

    return (
      <div className="grid grid__results">
        {this.props.movies.map(movie => {
          return movie.id && movie.poster_path && movie.title ? (
            <div className="grid__container" key={movie.id}>
              <img
                className="image-item"
                onClick={() => this.goToDetails(movie.id)}
                src={poster_path + movie.poster_path}
                style={{ height: '278px', width: '185px' }}
                alt="poster"
              />

              <div
                style={{
                  fontSize: '16px',
                  width: '185px',
                  margin: 'auto'
                }}
              >
                {movie.title}
              </div>
            </div>
          ) : null;
        })}
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => {
  return {
    onFetchMovieDetails: movieID => dispatch(fetchMovieDetails(movieID))
  };
};

export default connect(
  null,
  mapDispatchToProps
)(Gallery);
