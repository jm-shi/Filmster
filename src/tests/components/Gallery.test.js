import React from 'react';
import { shallow } from 'enzyme';
import Gallery from '../../components/Gallery';

describe('Gallery component', () => {
  let mockHistory, movies, wrapper;

  beforeEach(() => {
    mockHistory = { push: jest.fn() };
    movies = [
      {
        id: 123,
        poster_path: 'posterpath',
        title: 'A movie title'
      }
    ];
    wrapper = shallow(<Gallery movies={movies} history={mockHistory} />);
  });

  it('should render Gallery', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call history.push on click', () => {
    wrapper.find('.image-item').simulate('click', 'movieID');
    expect(mockHistory.push).toBeCalled();
  });
});
