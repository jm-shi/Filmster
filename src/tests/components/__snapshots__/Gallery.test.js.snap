// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery component should render Gallery 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Gallery
    history={
        Object {
            "push": [Function],
          }
    }
    movies={
        Array [
            Object {
              "id": 123,
              "poster_path": "posterpath",
              "title": "A movie title",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="grid__container"
>
          <img
                    alt="poster"
                    className="image-item"
                    onClick={[Function]}
                    src="https://image.tmdb.org/t/p/w185/posterpath"
                    style={
                              Object {
                                        "height": "278px",
                                        "width": "185px",
                                      }
                    }
          />
          <div
                    style={
                              Object {
                                        "fontSize": "16px",
                                        "margin": "auto",
                                        "width": "185px",
                                      }
                    }
          >
                    A movie title
          </div>
</div>,
      ],
      "className": "grid grid__results",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "123",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="poster"
              className="image-item"
              onClick={[Function]}
              src="https://image.tmdb.org/t/p/w185/posterpath"
              style={
                            Object {
                                          "height": "278px",
                                          "width": "185px",
                                        }
              }
/>,
            <div
              style={
                            Object {
                                          "fontSize": "16px",
                                          "margin": "auto",
                                          "width": "185px",
                                        }
              }
>
              A movie title
</div>,
          ],
          "className": "grid__container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "poster",
              "className": "image-item",
              "onClick": [Function],
              "src": "https://image.tmdb.org/t/p/w185/posterpath",
              "style": Object {
                "height": "278px",
                "width": "185px",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "A movie title",
              "style": Object {
                "fontSize": "16px",
                "margin": "auto",
                "width": "185px",
              },
            },
            "ref": null,
            "rendered": "A movie title",
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="grid__container"
>
            <img
                        alt="poster"
                        className="image-item"
                        onClick={[Function]}
                        src="https://image.tmdb.org/t/p/w185/posterpath"
                        style={
                                    Object {
                                                "height": "278px",
                                                "width": "185px",
                                              }
                        }
            />
            <div
                        style={
                                    Object {
                                                "fontSize": "16px",
                                                "margin": "auto",
                                                "width": "185px",
                                              }
                        }
            >
                        A movie title
            </div>
</div>,
        ],
        "className": "grid grid__results",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "123",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="poster"
                className="image-item"
                onClick={[Function]}
                src="https://image.tmdb.org/t/p/w185/posterpath"
                style={
                                Object {
                                                "height": "278px",
                                                "width": "185px",
                                              }
                }
/>,
              <div
                style={
                                Object {
                                                "fontSize": "16px",
                                                "margin": "auto",
                                                "width": "185px",
                                              }
                }
>
                A movie title
</div>,
            ],
            "className": "grid__container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "poster",
                "className": "image-item",
                "onClick": [Function],
                "src": "https://image.tmdb.org/t/p/w185/posterpath",
                "style": Object {
                  "height": "278px",
                  "width": "185px",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "A movie title",
                "style": Object {
                  "fontSize": "16px",
                  "margin": "auto",
                  "width": "185px",
                },
              },
              "ref": null,
              "rendered": "A movie title",
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
